components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,316 @@
+use std::{
+    char::{from_u32, MAX},
+    convert::TryFrom,
+    ops::RangeBounds,
+    slice::Iter,
+};
+
+use super::USetError;
+use crate::utils::{deconstruct_range, is_valid};
+/// Represents the end code point of the Basic Multilingual Plane range, starting from code point 0 , inclusive
+const BMP_MAX: u32 = 0xFFFF;
+
+/// UnicodeSet membership wrapper
+///
+/// Provides exposure to membership functions and constructors from serialized UnicodeSets
+/// and predefined ranges.
+/// Implements an [inversion list.](https://en.wikipedia.org/wiki/Inversion_list)
```

Comment:
I would move the inversion list sentence to the `inv_list` field. The internal representation seems like an implementation detail.

Reply:
agreed and changed